Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift

```diff
@@ -62,7 +62,7 @@ public class AWSClientConfiguration<ServiceSpecificConfiguration: AWSServiceSpec
     /// The credentials provider to be used for AWS credentials.
     ///
     /// If no credentials provider is supplied, the SDK will look for credentials in the environment, then in the `~/.aws/credentials` file.
-    public var credentialsProvider: CredentialsProviding
+    public var credentialsProvider: any CredentialsProviding
```

Comment:
Is there any way to avoid using an existential here?

i.e. could the concrete type of `CredentialsProviding` be known at compile time?